src/container/srv_container.c

```diff
@@ -1361,18 +1430,41 @@ cont_open(struct rdb_tx *tx, struct ds_pool_hdl *pool_hdl, struct cont *cont,
 	}
 
 	/* Determine pool meets container redundancy factor requirements */
-	if (!(in->coi_flags & DAOS_COO_FORCE)) {
+	is_healthy = cont_status_is_healthy(prop, &stat_pm_ver);
+	if (is_healthy) {
+		int	rf;
+
+		rf = daos_cont_prop2redunfac(prop);
 		pmap = pool_hdl->sph_pool->sp_map;
-		rc = cont_verify_redun_req(pmap, prop);
-		if (rc != 0) {
-			D_ERROR(DF_CONT": Container does not meet redundancy "
-					"requirements, set DAOS_COO_FORCE to "
-					"force container open rc: %d.\n",
-				DP_CONT(cont->c_svc->cs_pool_uuid,
-					cont->c_uuid), rc);
-			daos_prop_free(prop);
-			D_GOTO(out, rc);
+		rc = pool_map_rf_verify(pmap, stat_pm_ver, rf);
+		if (rc == -DER_RF) {
+			rc1 = cont_status_set_unclean(tx, pool_hdl->sph_pool,
+						      cont, prop);
+			if (rc1 != 0) {
+				D_ERROR(DF_CONT":set_unclean failed, "DF_RC"\n",
+					DP_CONT(cont->c_svc->cs_pool_uuid,
+						cont->c_uuid), DP_RC(rc1));
+				daos_prop_free(prop);
+				D_GOTO(out, rc1);
+			}
+
+			if (!(in->coi_flags & DAOS_COO_FORCE))
+				rf_fail = true;
+		} else if (rc) {
+			goto out;
 		}
+	} else if (!(in->coi_flags & DAOS_COO_FORCE)) {
+		rf_fail = true;
+	}
+	if (rf_fail) {
```

Comment:
[Nit-picking] rt_fail really necessary? We could remove the two FORCE checks above and change the rf_fail check to

    if (!(...flags & ...FORCE) {

Reply:
ok, I'll change it a little bit. thx

Reply:
Thanks.